rocketpy/EnvironmentAnalysis.py

```diff
@@ -605,12 +606,84 @@ def calculate_average_wind_profile(self):
         """average, 1, 2, 3 sigma wind profile from 0 35,000 ft AGL"""
         ...
 
-    # TODO: Implement
-    def calculate_average_day_wind_rose(self):
-        """average day wind rose"""
-        ...
+    def animate_average_wind_rose(self, figsize=(8, 8)):
+        """Animates the wind_rose of an average day. The inputs of a wind_rose are the location of the
+        place where we want to analyse, (x,y,z). The data is ensembled by hour, which means, the windrose
+        of a specific hour is generated by bringing together the data of all of the days available for that
+        specific hour. It's possible to change the size of the gif using the parameter figsize, which is the
+        height and width in inches.
+
+        Parameters
+        ----------
+        figsize : array
+
+        Returns
+        -------
+        Image : ipywidgets.widgets.widget_media.Image
+        """
+        days = list(self.surfaceDataDict.keys())
+        hours = list(self.surfaceDataDict[days[0]].keys())
+
+        windSpeed = {}
+        windDir = {}
+
+        for hour in hours:
+            windSpeed[hour] = []
+            windDir[hour] = []
+            for day in days:
+                windSpeed[hour].append(
+                    self.pressureLevelDataDict[day][hour]["windSpeed"](self.elevation)
+                )
+                windDir[hour].append(
+                    self.pressureLevelDataDict[day][hour]["windDirection"](
+                        self.elevation
+                    )
+                )
+
+        fig = plt.figure(
+            facecolor="w", edgecolor="w", figsize=figsize
+        )  # initialise la figure
+
+        filename = "wind_rose.gif"
+        FFMpegWriter = matplotlib.animation.writers["ffmpeg"]
```

Comment:
This line means that ffmpeg must be installed, correct?

Reply:
https://matplotlib.org/3.5.0/api/_as_gen/matplotlib.animation.FFMpegWriter.html

Reply:
Yes I think so, I would have to check if that would work on the collab

Reply:
I guess you would need to run something like `!sudo apt install ffmpeg`.

To be honest, I am not sure this is a good requirement to have, specially since it leaves outsite Python's ecosystem. Installation may be simple on Linux and Mac, but not so much on Windows.

We can discuss if there are any alternatives.

Reply:
Agreed, there are other writers, I'm going to check them